### PR TITLE
Add RFC: RawKV Stale Read

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Andy Lok <andylokandy@hotmail.com>

## Summary

Allow eventually consistency on RawKV.

## Motivation

Eventually consistency is acceptable in some scenarios when using RawKV. In this case, RawKV is allowed to break linear consistency and therefore improve the cluster-wise throughput.

[Rendered](https://github.com/andylokandy/rfcs/blob/staleread/text/0080-rawkv-stale-read.md)

### CODE REVIEW

**Comment by BusyJay** [comment]
What's the guarantees stale read provides? It seems there is no guarantee at all, client may read a key's value and then read no keys and then read the keys again even if there is no changes to the key.

**Comment by andylokandy** [comment]
@BusyJay This is exactly what eventually consistency means. To address these problems, the client should select the store with a satisfied RPO so that the freshness will not regress.

File: text/0080-rawkv-stale-read.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+### Implementation details
+
+1. In kvproto, add `direct_read` to `RaftRequestHeader`:
+
+    ```diff
+        // raft_cmdpb.proto
+        message RaftRequestHeader {
+            ...
+            bool replica_read = 8;
+    +       bool direct_read = 9;
```
Comment (BusyJay) [inline text/0080-rawkv-stale-read.md:50]: We already have the `STALE_READ` write batch flags, the new field may not be necessary.

**Pull request closed without merging**